internal/controller/etcdcluster_controller_test.go

```diff
@@ -30,55 +28,67 @@ import (
 	operatorv1alpha1 "go.etcd.io/etcd-operator/api/v1alpha1"
 )
 
-var _ = Describe("EtcdCluster Controller", func() {
-	Context("When reconciling a resource", func() {
-		const resourceName = "test-resource"
+func TestEtcdClusterController(t *testing.T) {
+	ctx := t.Context()
 
-		ctx := context.Background()
+	const resourceName = "test-etcd-cluster"
+	typeNamespacedName := types.NamespacedName{
+		Name:      resourceName,
+		Namespace: "etcd-operator-system",
+	}
 
-		typeNamespacedName := types.NamespacedName{
-			Name:      resourceName,
-			Namespace: "default", // TODO(user):Modify as needed
-		}
+	t.Run("When reconciling a resource", func(t *testing.T) {
 		etcdcluster := &operatorv1alpha1.EtcdCluster{}
 
-		BeforeEach(func() {
-			By("creating the custom resource for the Kind EtcdCluster")
-			err := k8sClient.Get(ctx, typeNamespacedName, etcdcluster)
-			if err != nil && errors.IsNotFound(err) {
-				resource := &operatorv1alpha1.EtcdCluster{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      resourceName,
-						Namespace: "default",
-					},
-					// TODO(user): Specify other spec details if needed.
-				}
-				Expect(k8sClient.Create(ctx, resource)).To(Succeed())
+		t.Log("Creating or retrieving the EtcdCluster resource")
```

Comment:
Small nit, at this step can we have some log message like "Checking for EtcdCluster resource"